titan-compiler/checker.lua

```diff
@@ -356,19 +368,26 @@ check_stat = function(stat, rettypes)
                 stat.decl.name)
         end
 
-        checkmatch("'for' start expression",
-            stat.decl._type, stat.start._type, stat.start.loc)
-        checkmatch("'for' finish expression",
-            stat.decl._type, stat.finish._type, stat.finish.loc)
-        checkmatch("'for' step expression",
-            stat.decl._type, stat.inc._type, stat.inc.loc)
+        checkmatch(stat.start.loc,
+            stat.decl._type, stat.start._type,
+            "'for' start expression")
```

Comment:
In repeat and while we have `<stat> statement ...` and here we have `'<stat>' ...`. I think we could chose one and have a standard.

Reply:
Sure... But I couldn't think of a good way to rewrite these for the "for". Any suggestions?

Reply:
We could use `''` in all of them.